application-collabora-default/src/main/java/com/xwiki/collabora/internal/configuration/DefaultCollaboraConfiguration.java

```diff
@@ -41,13 +43,32 @@
 @Singleton
 public class DefaultCollaboraConfiguration implements CollaboraConfiguration
 {
+    private static final String SERVER = "server";
+
+    @Inject
+    @Named(MainCollaboraConfigurationSource.HINT)
+    private ConfigurationSource mainConfiguration;
+
     @Inject
-    @Named("collabora")
-    private ConfigurationSource configuration;
+    @Named(CollaboraConfigurationSource.HINT)
+    private ConfigurationSource currentConfiguration;
 
     @Override
     public URL getDiscoveryURL() throws MalformedURLException
     {
-        return new URL(this.configuration.getProperty("server") + "/hosting/discovery");
+
+        return new URL(getServerConfiguration() + "/hosting/discovery");
+    }
+
+    /**
+     * Fallback on the main wiki server configuration in case it was not defined at the wiki level.
+     *
+     * @return the Collabora Online server defined in configuration
+     */
+    private String getServerConfiguration()
+    {
+        String currentWikiServer = this.currentConfiguration.getProperty(SERVER);
+        String mainWikiServer = this.mainConfiguration.getProperty(SERVER);
+        return StringUtils.isEmpty(currentWikiServer) ? mainWikiServer : currentWikiServer;
```

Comment:
You could avoid looking for the main wiki configuration when the subwiki specifies the Collabora server URL:

```suggestion
        return StringUtils.isEmpty(currentWikiServer) ? this.mainConfiguration.getProperty(SERVER) : currentWikiServer;
```